tooling/ast_fuzzer/src/compare.rs

```diff
@@ -164,6 +211,78 @@ impl From<(SsaProgramArtifact, CompareOptions)> for CompareArtifact {
     }
 }
 
+/// Compare the execution of a Noir program in pure comptime (via interpreter)
+/// vs normal SSA execution.
+pub struct CompareComptime {
+    pub program: Program,
+    pub abi: Abi,
+    pub input_map: InputMap,
+    pub source: String,
+    pub ssa: CompareArtifact,
+    pub force_brillig: bool,
+}
+
+impl CompareComptime {
+    /// Execute the Noir code and the SSA, then compare the results.
+    pub fn exec(&self) -> eyre::Result<CompareResult> {
+        let program1 = match prepare_and_compile_snippet(self.source.clone(), self.force_brillig) {
+            Ok((program, _)) => program,
+            Err(e) => panic!("failed to compile program:\n{}\n{e:?}", self.source),
+        };
+
+        let blackbox_solver = Bn254BlackBoxSolver(false);
+        let initial_witness = self.abi.encode(&self.input_map, None).wrap_err("abi::encode")?;
+
+        let do_exec = |program| {
+            let mut print = Vec::new();
+
+            let mut foreign_call_executor = DefaultForeignCallBuilder::default()
+                .with_mocks(false)
+                .with_output(&mut print)
+                .build();
+
+            let res = nargo::ops::execute_program(
+                program,
+                initial_witness.clone(),
+                &blackbox_solver,
+                &mut foreign_call_executor,
+            );
+
+            let print = String::from_utf8(print).expect("should be valid utf8 string");
+            (res, print)
+        };
+
+        let (res1, print1) = do_exec(&program1.program);
+        let (res2, print2) = do_exec(&self.ssa.artifact.program);
+
+        CompareResult::new(&self.abi, (res1, print1), (res2, print2))
+    }
+
+    /// Generate a random comptime-viable AST, reverse it into
+    /// Noir code and also compile it into SSA.
+    pub fn arb(
+        u: &mut Unstructured,
+        c: Config,
+        f: impl FnOnce(
+            &mut Unstructured,
+            Program,
+        ) -> arbitrary::Result<(SsaProgramArtifact, CompareOptions)>,
+    ) -> arbitrary::Result<Self> {
+        let force_brillig = c.force_brillig;
+        let program = arb_program_comptime(u, c)?;
+        let abi = program_abi(&program);
+
+        let ssa = CompareArtifact::from(f(u, program.clone())?);
+
+        let input_program = &ssa.artifact.program;
+        let input_map = arb_inputs(u, input_program, &abi)?;
```

Comment:
Now that we're not printing the inputs in the results, I'd remove even these if we really don't have any inputs.